File: Leetcode/python/sol349.py
# https://leetcode.com/problems/intersection-of-two-arrays/description/
from collections import defaultdict, Counter
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1_set = set(nums1)
        nums2_set = set(nums2)
        return list(nums1_set & nums2_set)
        # version 2
        # all_values = list(set(nums1)) + list(set(nums2))
        # count = Counter(all_values)
        # result = []
        # for val, freq in count.items():
        #     if freq > 1:
        #         result.append(val)
        # return result
        # version 2
        # nums1_set = set(nums1)
        # nums2_set = set(nums2)
        # counter = defaultdict(int)
        # for i in nums1_set:
        #     counter[i]+=1
        # for i in nums2_set:
        #     counter[i]+=1
        # return [i for i, j in counter.items() if j > 1]

File: Leetcode/cpp/sol349.cpp

//https://leetcode.com/problems/intersection-of-two-arrays/submissions/1658507305/

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {

        set<int> nums1_set(nums1.begin(), nums1.end());
        set<int> nums2_set(nums2.begin(), nums2.end());

        set<int> result;
        set_intersection(nums1_set.begin(), nums1_set.end(),
                        nums2_set.begin(), nums2_set.end(),
                        inserter(result, result.begin()));
        vector <int> rslt(result.begin(), result.end());

        return rslt;
    }
};